flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -234,30 +234,42 @@ struct Evaluation : EvaluationVariant {
     return visit(common::visitors{
         [](auto &r) { return pft::isDirective<std::decay_t<decltype(r)>>; }});
   }
+  constexpr bool isNopConstructStmt() const {
+    return visit(common::visitors{[](auto &r) {
+      return pft::isNopConstructStmt<std::decay_t<decltype(r)>>;
+    }});
+  }
+
   /// Return the predicate:  "This is a non-initial, non-terminal construct
   /// statement."  For an IfConstruct, this is ElseIfStmt and ElseStmt.
   constexpr bool isIntermediateConstructStmt() const {
     return visit(common::visitors{[](auto &r) {
       return pft::isIntermediateConstructStmt<std::decay_t<decltype(r)>>;
     }});
   }
-  constexpr bool isNopConstructStmt() const {
-    return visit(common::visitors{[](auto &r) {
-      return pft::isNopConstructStmt<std::decay_t<decltype(r)>>;
-    }});
-  }
 
-  /// Return nested evaluation list that may be empty.
-  EvaluationList &getNestedEvaluations() {
-    assert(evaluationList && "no nested evaluations");
-    return *evaluationList;
+  /// Return the first non-nop successor of an evaluation, possibly exiting
+  /// from one or more enclosing constructs.
+  Evaluation *nonNopSuccessor() const {
```

Comment:
Since there is an assert that the returned `Evaluation*` cannot be a `nullptr`, could this return and `Evaluation&` instead ?